beesdoo_shift_swap/models/planning.py

```diff
@@ -32,23 +32,24 @@ def _prepare_task_day(self):
         )
 
         changes = (
-            set(swaps)
-            | set(exchanges)
-            | set(solidarity_offers)
-            | set(solidarity_requests)
+            list(swaps)
+            + list(exchanges)
+            + list(solidarity_offers)
+            + list(solidarity_requests)
         )
-        processed = set()
+
+        # Sort changes by validation date to evaluate them in the correct order
+        changes.sort(key=lambda x: x.get_validate_date())
 
         swap_subscription_done = []
         for shift in shifts:
-            # Sort changes by validation date to evaluate them in the correct order
-            to_process = sorted(changes, key=lambda x: x.get_validate_date())
-            for rec in to_process:
-                shift, swap_subscription_done, done = rec.update_shift_data(
-                    shift, swap_subscription_done
-                )
-                if done:
-                    processed.add(rec)
-            changes -= processed
+            # process the remaining changes and update the list to contain
+            # only the changes that have not be processed yet.
+            changes[:] = (
```

Comment:
It is not clear to me if `[:]` is needed. Can't you do a list comprehension instead of a generator? i.e. replace `(rec for rec in ...)` with `[rec for rec in ...]`

Reply:
you’re right. this makes sense only if the list is really big and making a copy of it should be avoided. i’ve changed it.